src/application_input.hpp

```diff
@@ -33,7 +33,7 @@ struct ApplicationInput {
   std::function<Packages_t(std::unique_ptr<ParameterInput> &)> ProcessPackages = nullptr;
 
   // Mesh functions
-  std::function<void(ParameterInput *)> InitUserMeshData = nullptr;
+  std::function<void(Mesh *, ParameterInput *)> InitUserMeshData = nullptr;
```

Comment:
Is this change still required?

Reply:
This isn't a required change, but from my earlier discarded attempt at this PR I was left with the impression that `InitUserMeshData` basically isn't useful unless we pass the `Mesh` pointer in so I figured I would leave this change in case this function is needed in the future -- is that reasonable?

Reply:
Agreed, this changes makes sense so let's just keep it here.